arch/risc-v/src/common/riscv_internal.h

```diff
@@ -311,6 +311,14 @@ static inline uintptr_t *riscv_vpuregs(struct tcb_s *tcb)
 #  define riscv_vpuregs(tcb)
 #endif
 
+/* IRQ Flag */
+
+noinstrument_function
+static inline_function void riscv_set_interrupt_context(bool flag)
```

Comment:
We have a special interrupt controller like riscv RNMI that wants to use this api, but this part is not implemented in the chip. so, @xiaoxiang781216 Don't you think this design is strange? The interrupt status can only be updated by the chip implementation, why?

Reply:
so you want to update irq status from application? it's strange to me too. riscv_set_interrupt_context can be called from arch or chip specific code, I think that it's enough to cover your special need.

Reply:
> So do you want updating irq status from application?

Not update from application, this is common driver used in board-level implementation. This driver will be used by MCUs of 3 architectures: arm/arm64/riscv. More detail, it is a peripheral monitor that plugs into all MCUs, Like nvidia FSI( Functional Safety Island):
https://developer.nvidia.com/docs/drive/drive-os/6.0.9/public/drive-os-linux-sdk/common/topics/fsi_integration/Functional_Safety_Island.html

Reply:
ok